### PR TITLE
Add utilities for bazel client version parsing and comparison

### PR DESCRIPTION
**Pull request by bduffany**
Context: https://github.com/buildbuddy-io/buildbuddy/pull/2916#issuecomment-1314209533

This will also be useful to address https://github.com/buildbuddy-io/buildbuddy-internal/issues/1187

---

**Version bump**: Patch

### CODE REVIEW

File: server/util/bazel_request/bazel_request.go [hunk 1]
```diff
@@ -33,6 +41,89 @@ func GetInvocationID(ctx context.Context) string {
...
+			return nil
+		}
+	}
+	b.Suffix = m[bazelVersionPattern.SubexpIndex("suffix")]
+	return b
+}
+
+// IsAtLeast returns whether this bazel version is equal to or supercedes the
+// given version.
+func (v *Version) IsAtLeast(c *Version) bool {
```
Comment (tylerwilliams) [inline server/util/bazel_request/bazel_request.go:92]: I wonder if we should add a variant of this function that accepts a string version and returns (bool, err) instead, for ease of use / readability when we actually use this?

**Review by tylerwilliams — APPROVED** [review]
nice :)

File: server/util/bazel_request/bazel_request.go (hunk 1 above)
Comment (bduffany) [reply server/util/bazel_request/bazel_request.go:98]: ^ going to submit as-is, and we can decide later whether we need a util method

Comment (tylerwilliams) [reply server/util/bazel_request/bazel_request.go:98]: SGTM

**Pull request merged**